src/glossary.adoc

```diff
@@ -0,0 +1,116 @@
+[[sec:glossary]]
+[glossary]
+= Glossary
+
+[glossary]
+
+arity::
+The number of arguments declared by or passed to a macro.  A macro can be _fixed_ or _variable_
+arity; the number of subforms in an invocation is its _actual arity_.
+
+argument::
+A single subform within an encoding expression, corresponding to one of the macro’s parameters.
+This is a one-to-one relation except in the case of rest parameters, where any number of trailing
+arguments are passed to the same parameter.
+
+cardinality::
+The number of elements accepted by a single argument of a macro, or produced by a macro. The
+arguments and results of a macro are streams, and each can be constrained to have exactly-one
+(`*!*`), zero-or-one (`*?*`), zero-or-more (`*{asterisk}*`, `*\...*`), or one-or-more (`*+*`)
+elements.
+
+declaration::
+The association of a name with an entity (for example, a module or macro). See also _definition_. Not
+all declarations are definitions: some introduce new names for existing entities.
+
+definition::
+The specification of a new entity. Most (but not all) definitions are declarations.
+
+directive::
+A unit of data in an Ion document that affects the encoding context, and thus the way the
+document’s data is decoded. In Ion 1.0 there are two directives: the Ion Version Marker, and
+top-level `$ion_symbol_table` structs. Ion 1.1 adds the `$ion_encoding` directive.
```

Comment:
I contend that the IVM is not data, so I would not say a unit of data.  Maybe "syntax or unit of data?"

Reply:
I guess we need to define "data"!